rclcpp/include/rclcpp/loaned_message.hpp

```diff
@@ -184,13 +168,18 @@ class LoanedMessage
    * A call to `release()` will unmanage the memory for the ROS message.
    * That means that the destructor of this class will not free the memory on scope exit.
```

Comment:
We should update this documentation, and mention that the memory is free'd when the unique pointer goes out instead, and that the unique pointer will keep the publisher alive until it is destroyed.